gridcomps/History/MAPL_HistoryGridComp.F90

```diff
@@ -5170,5 +5180,41 @@ subroutine RecordRestart( gc, import, export, clock, rc )
     _RETURN(ESMF_SUCCESS)
   end subroutine RecordRestart
 
+  subroutine  checkIfStateHasField(state, fieldName, hasField, rc)
+    type(ESMF_State), intent(in) :: state ! export state
+    character(len=*), intent(in) :: fieldName
+    logical, intent(out)         :: hasField
+    integer, intent(out), optional :: rc ! Error code:
+
+    integer :: n, i, status
+    character (len=ESMF_MAXSTR), allocatable  :: itemNameList(:)
+    type(ESMF_StateItem_Flag),   allocatable  :: itemTypeList(:)
+
+    call ESMF_StateGet(state, itemcount=n,  rc=status)
+    _VERIFY(status)
+
+    allocate(itemNameList(n), stat=status)
+    _VERIFY(status)
+    allocate(itemTypeList(n), stat=status)
+    _VERIFY(status)
+    call ESMF_StateGet(state,itemnamelist=itemNamelist,itemtypelist=itemTypeList,rc=status)
+    _VERIFY(STATUS)
+
+    hasField = .false.
+    do I=1,N
+       if(itemTypeList(I)/=ESMF_STATEITEM_FIELD) cycle
+       if(itemNameList(I)==fieldName) then
+          hasField = .true.
+          exit
+       end if
+    end do
+    deallocate(itemNameList, stat=status)
+    _VERIFY(STATUS)
+    deallocate(itemTypeList, stat=status)
+    _VERIFY(status)
```

Comment:
Unnecessary.  Unsaved ALLOCATABLE's are automatically deallocated on exit.

Again - no change requested.   Just for the future.